### ISSUE TITLE
[BUG] DataLakePathClient corrupts specific storage names

### ISSUE THREAD

**Issue by vrozovmicrosoft** [Azure/azure-sdk-for-net#28386]
### Library name and version

Azure.Storage.Files.DataLake 12.8.0

### Describe the bug

DataLakePathClient renames isctordfsblobdfsblob storage account to isctorblobblobdfsblob when performs the call to the service.

### Expected behavior

Keeping the original storage account name

### Actual behavior

'dfs' substring in the storage name is replaced by 'blob'

### Reproduction Steps

var dataLakeFileSystemClient = new DataLakeFileSystemClient(new Uri([https://isctordfsblobdfsblob.dfs.core.windows.net](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fisctordfsblobdfsblob.dfs.core.windows.net%2F&data=05%7C01%7Cvrozov%40microsoft.com%7C9c51750bd4234d52e27e08da2760c034%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637865593748073972%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=jMukqS0vjWk6r9IlEwhM1g0m4oRW7iOptYVE%2BGTkN3k%3D&reserved=0)));
var pathClient = new DataLakePathClient(dataLakeFileSystemClient, "slvrcntr");
// At this point you can see that Uri inside pathClient is correct: https://isctordfsblobdfsblob.dfs.core.windows.net
pathClient.Exists();
The exception thrown from Exists() shows incorrect storage name ('dfs' replaced by 'blob'):

RequestFailedException: The remote name could not be resolved: 'isctorblobblobdfsblob.dfs.core.windows.net'

### Environment

.NET Framework 4.7.2

**Comment by SaurabhSharma-MSFT** [comment]
@vrozovmicrosoft We are looking into it and get back to you for any additional information.

**Comment by SaurabhSharma-MSFT** [comment]
@vrozovmicrosoft You need to pass the correct uri path like `"https://{StorageName}.file.core.windows.net/{ShareName}`.
For example - If the share is as below 
![image](https://user-images.githubusercontent.com/38112130/165648770-27b77a96-d737-402e-983d-a64a461b375c.png)

Then you need to pass the path like below to DatalakeFileSystemClient and pass the correct path to the file to DatalakePathClient.

```
var dataLakeFileSystemClient = new DataLakeFileSystemClient(new Uri("https://{StorageName}.file.core.windows.net/sales"), sharedKeyCredential);
var pathClient = new DataLakePathClient(dataLakeFileSystemClient, "2020/Sales-2.csv");
var exists = pathClient.Exists();
```

**Comment by vrozovmicrosoft** [comment]
Exactly the same problem reproduces if I add the container name to the Uri for DataLakeFileSystemClient.
The problem is specific for 'blob' / 'dfs' substring combinations.
We have found that the problem was fixed in version 12.9, see https://github.com/Azure/azure-sdk-for-net/blob/Azure.Storage.Files.DataLake_12.9.0/sdk/storage/Azure.Storage.Files.DataLake/CHANGELOG.md, but currently we cannot move to this version.
You can proceed with the issue closure.
Thanks.

**Comment by SaurabhSharma-MSFT** [comment]
@vrozovmicrosoft Ok. Got it.  Thanks.
We will now proceed to close this thread. If there are further questions regarding this matter, please reopen it and we will gladly continue the discussion.
